File: Excel/475 Challenge.py
import pandas as pd
import re

input = pd.read_excel("475 Split by Positions.xlsx", skiprows=1, usecols="A:B")
test = pd.read_excel("475 Split by Positions.xlsx", skiprows=1, usecols="C:H")

def split_string_by_pos(string, positions_str):
    positions = [int(pos) - 1 for pos in re.split(r"\s*,\s*", positions_str)]
    starts = [0] + positions
    ends = positions + [len(string)]
    return [string[start:end] for start, end in zip(starts, ends)]

result = input.copy()
result["split"] = result.apply(lambda row: split_string_by_pos(row["Names"], row["Position"]), axis=1)
result = result.drop("Position", axis=1).join(result["split"].apply(pd.Series))
result = result.drop(["Names", "split"], axis=1).reset_index(drop=True)
result.columns = test.columns

print(result.equals(test)) # True

File: Excel/400-499/475/475 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/475 Split by Positions.xlsx", range = "A2:B12")
test  = read_excel("Excel/475 Split by Positions.xlsx", range = "C2:H12")

split_string_by_pos <- function(string, positions_str) {
  positions <- str_split(positions_str, "\\s*,\\s*") %>% 
    unlist() %>% 
    as.numeric()
  starts <- c(1, positions)
  ends <- c(positions - 1, nchar(string))
  map2(starts, ends, ~ substr(string, .x, .y))
}

result = input %>%
  mutate(split = map2(Names, Position, split_string_by_pos)) %>%
  unnest_wider(split, names_sep = "_") %>%
  select(Text1 = split_1, Text2 = split_2, Text3 = split_3, Text4 = split_4, Text5 = split_5, Text6 = split_6)

identical(result, test)
# [1] TRUE